.paw/work/workflow-handoffs/ImplementationPlan.md

```diff
@@ -0,0 +1,669 @@
+
+The approach proceeds in logical phases: first establish handoff infrastructure (tool, WorkflowContext.md field), then implement Status Agent enhancements (state analysis, dynamic prompts, help mode), followed by agent instruction updates to use handoff capabilities, and finally comprehensive testing. Each phase is independently testable and builds on prior phases without breaking existing functionality.
+
+## Phase Summary
+
+1. **Phase 1: Handoff Tool Foundation** - Implement `paw_call_agent` tool, add Handoff Mode to WorkflowContext.md, establish handoff patterns
+2. **Phase 2: Status Agent Core Enhancements** - Add workflow state scanning, artifact detection, git status checking, PR querying
+3. **Phase 3: Dynamic Prompt Generation** - Implement `paw_generate_prompt` tool with phase-specific context injection
+4. **Phase 4: Agent Instruction Updates** - Update all agents to present next-step options and invoke handoff tool based on mode
+5. **Phase 5: Testing and Validation** - Comprehensive unit tests, integration tests, manual workflow testing
+
+---
+
+## Phase 1: Handoff Tool Foundation
+
+### Overview
+Establish the core handoff infrastructure by implementing the `paw_call_agent` tool that enables agent-to-agent transitions, adding Handoff Mode field to WorkflowContext.md, and creating the patterns agents will use to invoke handoffs.
+
+### Changes Required:
+
+#### 1. Handoff Tool Implementation
+**File**: `src/tools/handoffTool.ts` (new)
+**Changes**:
+- Create `HandoffParams` interface with fields: `target_agent` (string, required), `work_id` (string, required), `inline_instruction` (string, optional)
+- Implement validation: Work ID matches `/^[a-z0-9-]+$/`, target agent non-empty, inline instruction max 500 chars
+- Implement `determineAgentFromIdentifier()` function mapping friendly names to agent strings:
+  - `research` → `PAW-01B Spec Researcher`
+  - `code` → `PAW-02A Code Researcher`
+  - `plan` → `PAW-02B Impl Planner`
+  - `implement` or `implement Phase N` → `PAW-03A Implementer`
+  - `review` → `PAW-03B Impl Reviewer`
+  - `docs` → `PAW-04 Documenter`
+  - `pr` → `PAW-05 PR`
+  - `status` → `PAW-X Status Update`
+- Implement prerequisite validation `validateHandoffPrerequisites()`:
+  - Check WorkflowContext.md exists in `.paw/work/<work_id>/`
+  - For `plan`, verify Spec.md exists
+  - For `implement`, verify ImplementationPlan.md exists
+  - For `implement Phase N`, verify `## Phase N` heading exists in ImplementationPlan.md
+  - Return actionable error messages on failure
+- Implement inline instruction parsing: extract text after "but", "with", "remember to" keywords
+- Construct prompt: frontmatter with agent name, instruction for stage, `Work ID: <work_id>`, appended inline instruction if provided
```

Comment:
Frontmatter is not used for the prompt in workbench.action.chat.open. The 'mode' parameter is used with the agent name. Ensure 'mode' is the proper name for the parameter, perhaps it was update in the API - we might want to up the version of the vscode API that is brought in (which might restrict teh min version of VS Code that we support, which should be documented in the README).